fenwick/src/main/java/org/opencadc/fenwick/Main.java

```diff
@@ -1,114 +1,247 @@
+            final String dbUrl = Main.readProperty(props, DB_URL_CONFIG_KEY, errorMessage);
+
+            daoConfig.put("schema", Main.readProperty(props, DB_SCHEMA_CONFIG_KEY, errorMessage));
+            daoConfig.put("database", "inventory");
+
+            if (StringUtil.hasText(username) && StringUtil.hasText(password) && StringUtil.hasText(dbUrl)) {
+                final ConnectionConfig cc = new ConnectionConfig(null, null, username, password,
+                                                                 "org.postgresql.Driver", dbUrl);
+
+                try {
+                    DBUtil.createJNDIDataSource("jdbc/inventory", cc);
+                } catch (NamingException ne) {
+                    throw new IllegalStateException(String.format("Unable to access database: %s", dbUrl), ne);
+                }
+
+                daoConfig.put("jndiDataSourceName", "jdbc/inventory");
+            }
+
+            final String configuredSQLGenerator = Main.readProperty(props, SQLGENERATOR_CONFIG_KEY, errorMessage);
+            if (StringUtil.hasText(configuredSQLGenerator)) {
+                daoConfig.put(SQLGENERATOR_CONFIG_KEY, Class.forName(configuredSQLGenerator));
+            }
+
+            final String configuredQueryService = Main.readProperty(props, QUERY_SERVICE_CONFIG_KEY, errorMessage);
+            final URI resourceID = StringUtil.hasText(configuredQueryService)
+                                   ? URI.create(configuredQueryService)
+                                   : null;
+
+            final String configuredArtifactSelector = Main.readProperty(props, ARTIFACT_SELECTOR_CONFIG_KEY,
+                                                                        errorMessage);
+            final ArtifactSelector selector = StringUtil.hasText(configuredArtifactSelector)
+                                              ? InventoryUtil.loadPlugin(configuredArtifactSelector)
+                                              : null;
+
+            // Missing required elements.  Only the expectedFilters and logging are optional.
+            if (errorMessage.length() > 0) {
+                System.err.println(
+                        String.format("\nConfiguration file %s missing one or more values: %s.\n", CONFIG_FILE_NAME,
+                                      errorMessage.toString()));
+                System.exit(2);
+            }
+
             InventoryHarvester doit = new InventoryHarvester(daoConfig, resourceID, selector);
             doit.run();
             System.exit(0);
         } catch (Throwable unexpected) {
-            log.error("unexpected failure", unexpected);
+            log.error("Unexpected failure", unexpected);
+            System.err.println("Unexpected failure.  Check log output.");
             System.exit(-1);
         }
     }
-    
-    private Main() { 
+
+    /**
+     * Read in the configuration file and load it into the given Properties.
+     */
```

Comment:
this method not needed -- applications don't encode the "find config file" logic